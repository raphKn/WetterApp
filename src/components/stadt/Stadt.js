import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, View } from 'react-native';
import CustomText from '../common/CustomText';
import CustomSection from '../common/CustomSection';
import { isEmptyObject } from '../../utils/ObjectUtils';

const styles = StyleSheet.create({
    container: {},
    keyValue: {
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
    stadtName: {
        fontSize: 18,
    },
});

const Stadt = (props) => {
    if (isEmptyObject(props.stadt)) {
        return null;
    }
    return (<CustomSection style={styles.container} name="Stadt">
        <View style={styles.keyValue}>
            <CustomText style={styles.stadtName}>Stadt</CustomText>
            <CustomText style={styles.stadtName}>{props.stadt.stadtName}</CustomText>
        </View>
        <View style={styles.keyValue}>
            <CustomText>Longitude</CustomText>
            <CustomText>{props.stadt.longitude}</CustomText>
        </View>
        <View style={styles.keyValue}>
            <CustomText>Latitude</CustomText>
            <CustomText>{props.stadt.latitude}</CustomText>
        </View>
    </CustomSection>);
};

Stadt.propTypes = {
    stadt: PropTypes.shape({
        stadtName: PropTypes.string,
        longitude: PropTypes.number,
        latitude: PropTypes.number,
    }),
};

Stadt.defaultProps = {
    stadt: {},
};

export default Stadt;
