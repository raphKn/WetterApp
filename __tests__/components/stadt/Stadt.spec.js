import React from 'react';
import renderer from 'react-test-renderer';
import Stadt from '../../../src/components/stadt/Stadt';

describe('Stadt', () => {
    const stadt = {
        stadtName: 'name',
        longitude: 20.01,
        latitude: 20.02,
    };
    it('renders maxTemp correctly in a snapshot', () => {
        const wrapper = renderer.create(
            <Stadt stadt={stadt} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('renders nothing for no stadt in a snapshot', () => {
        const wrapper = renderer.create(
            <Stadt stadt={{}} />);
        expect(wrapper).toMatchSnapshot();
    });
});
