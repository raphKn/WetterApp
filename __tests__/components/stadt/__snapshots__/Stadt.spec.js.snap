// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stadt renders maxTemp correctly in a snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "paddingTop": 12,
      },
      Object {
        "backgroundColor": "#3770a5",
      },
    ]
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#FAFAFA",
          },
          Object {
            "fontSize": 18,
            "paddingHorizontal": 18,
          },
        ]
      }
    >
      Stadt
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#E0E0E0",
          "paddingHorizontal": 18,
          "paddingVertical": 6,
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#212121",
            },
            Object {
              "fontSize": 18,
            },
          ]
        }
      >
        Stadt
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#212121",
            },
            Object {
              "fontSize": 18,
            },
          ]
        }
      >
        name
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#212121",
            },
            null,
          ]
        }
      >
        Longitude
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#212121",
            },
            null,
          ]
        }
      >
        20.01
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#212121",
            },
            null,
          ]
        }
      >
        Latitude
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#212121",
            },
            null,
          ]
        }
      >
        20.02
      </Text>
    </View>
  </View>
</View>
`;

exports[`Stadt renders nothing for no stadt in a snapshot 1`] = `null`;
